diff --git a/lib/calib/calib_basic.c b/lib/calib/calib_basic.c
index 9ee5eabf..0bc6925d 100644
--- a/lib/calib/calib_basic.c
+++ b/lib/calib/calib_basic.c
@@ -44,6 +44,7 @@ ATCA_STATUS calib_wakeup_i2c(ATCADevice device)
     {
         int retries = atca_iface_get_retries(iface);
         uint8_t address = atcab_get_device_address(device);
+        iface->mIfaceCFG->atcai2c.address = 0x00;
         uint32_t temp;
         uint32_t wake;
         uint16_t rxlen;
@@ -82,6 +83,7 @@ ATCA_STATUS calib_wakeup_i2c(ATCADevice device)
     #endif
             atca_delay_us(atca_iface_get_wake_delay(iface));
 
+            iface->mIfaceCFG->atcai2c.address = address;
             rxlen = (uint16_t)sizeof(wake);
             if (ATCA_SUCCESS == status)
             {
