From 9ffb6f8e097a4fd1b53c8dcaceba36d5eb4f5956 Mon Sep 17 00:00:00 2001
From: Matt Wood <matt.wood@microchip.com>
Date: Mon, 8 Jan 2024 16:50:10 -0500
Subject: [PATCH 1/2] add initramfs

Signed-off-by: Matt Wood <matt.wood@microchip.com>
---
 sam9x60_curiosity.its | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/sam9x60_curiosity.its b/sam9x60_curiosity.its
index e9cd9fd..f941e3e 100644
--- a/sam9x60_curiosity.its
+++ b/sam9x60_curiosity.its
@@ -45,6 +45,18 @@
 			};
 		};
 
+		initramfs {
+			description = "system initramfs";
+			data = /incbin/("./initramfs-image-sam9x60-curiosity-sd.cpio.gz");
+			type = "ramdisk";
+			arch = "ARM";
+			os = "linux";
+			compression = "none";
+			hash-1 {
+				algo = "sha256";
+			};
+		};
+
 		fdt_pda5 {
 			description = "Device Tree blob pda5000 overlay";
 			data = /incbin/("./sam9x60_curiosity/sam9x60_curiosity_pda5.dtbo");
@@ -85,6 +97,7 @@
 			description = "Linux kernel and base FDT blob for SAM9X60-Curiosity board";
 			kernel = "kernel";
 			fdt = "base_fdt";
+			ramdisk = "initramfs";
 		};
 
 		base_dtb {
-- 
2.34.1

